packages/shared-pure/src/types/ValueType.ts

```diff
@@ -0,0 +1,17 @@
+export interface ValueType extends String {
+  _ValueTypeBrand: string
+}
+
+export function ValueType(value: string): ValueType {
+  if (!TYPES.includes(value)) {
+    throw new TypeError('Invalid ValueType')
+  }
+  return value as unknown as ValueType
+}
+
+const TYPES = ['TVL', 'CBV', 'EBV', 'NMV']
+
+ValueType.TVL = ValueType(TYPES[0])
```

Comment:
Maybe leave a comment here that explains what TVL, CBV, EBV and NMV actually stand for?